<launch>
  <arg name="camera_name" default="thermal"/>

  <node pkg="nodelet" type="nodelet" name="flir_nodelet_manager"  args="manager" 
    output="screen"/>

  <node pkg="nodelet" type="nodelet" name="flir_nodelet1" 
    args="load flir_ros_sync/FlirRos flir_nodelet_manager" output="screen">
        <param name="device_name" value="/dev/video0"/>
        <param name="raw" value="true" type="bool" />
        <param name="publish_image_sharing_every_n" value="2" type="int" />
        <param name="send_every_n" value="1" type="int" />

        <param name="use_ext_sync" value="1" type="int" />

        <param name="width" value="640" type="int"/>
        <param name="height" value="512" type="int"/>

        <param name="camera_name" value="$(arg camera_name)"/>
        <param name="intrinsic_url" value="package://flir_ros_sync/data/camera_info/example/thermal_caminfo_flir_640.yaml"/>
        <remap from="/set_camera_info" to="$(arg camera_name)/set_camera_info"/>

	<!-- 0 high, 1 low, 2 auto, 3 dual, 4 manual -->
	<param name="gain_mode" value="2" type="int" />

	<!-- 0 manual, 1 auto, 2 external, 3 shutter test -->
	<param name="ffc_mode" value="1" type="int" />

	<!-- 0 disable, 1 master, 2 slave -->
	<param name="sync_mode" value="1" type="int" />
	
	<param name="set_config" value="true" type="bool" />
  </node>

  <node pkg="nodelet" type="nodelet" name="flir_nodelet2" 
    args="load flir_ros_sync/FlirRos flir_nodelet_manager" output="screen">
        <param name="device_name" value="/dev/video2"/>
        <param name="raw" value="true" type="bool" />
        <param name="publish_image_sharing_every_n" value="2" type="int" />
        <param name="send_every_n" value="1" type="int" />

        <param name="use_ext_sync" value="1" type="int" />

        <param name="width" value="640" type="int"/>
        <param name="height" value="512" type="int"/>

        <param name="camera_name" value="$(arg camera_name)2"/>
        <param name="intrinsic_url" value="package://flir_ros_sync/data/camera_info/example/thermal_caminfo_flir_640.yaml"/>
        <remap from="/set_camera_info" to="$(arg camera_name)/set_camera_info2"/>

	<!-- 0 high, 1 low, 2 auto, 3 dual, 4 manual -->
	<param name="gain_mode" value="2" type="int" />

	<!-- 0 manual, 1 auto, 2 external, 3 shutter test -->
	<param name="ffc_mode" value="1" type="int" />

	<!-- 0 disable, 1 master, 2 slave -->
	<param name="sync_mode" value="1" type="int" />
	
	<param name="set_config" value="false" type="bool" />
  </node>

</launch>
